src/App.js

```diff
@@ -1,20 +1,48 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import TodoList  from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+const useSemiPersistentState = () => {
+
+  
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')));
```

Comment:
Same here as lesson 1.6. Make sure that use state gets a default empty array value if the localStorage happens to be empty.